### PR TITLE
Remove `ValidatorApiHandlerGloas`

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
Moving to a world where we wouldn't have an execution payload state, so maintaning fork specific `ValidatorApiHandler` is not necessary

related to #10562 

## Fixed Issue(s)
N/A

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Medium Risk**
> Unifies validator API logic across milestones and changes how attestation committee indices and chain-head state are derived, which can affect validator duties/attestation correctness and head updates in edge cases.
> 
> **Overview**
> Removes the milestone-specific `ValidatorApiHandlerGloas` and updates all wiring/tests to always construct `ValidatorApiHandler`, extending its constructor to accept `ExecutionPayloadBidManager` and `ProposerPreferencesManager`.
> 
> `ValidatorApiHandler` now implements Gloas endpoints directly: `sendSignedProposerPreferences` is validated via `ProposerPreferencesManager`, and `publishSignedExecutionPayloadBid` validates/records bids via `ExecutionPayloadBidManager` (throwing on rejection). Attestation production now computes the `AttestationData` committee index via `ForkChoiceUtil.computeCommitteeIndexForAttestation(...)` (with a Gloas override in `ForkChoiceUtilGloas`), and block-production preparation uses `CombinedChainDataClient.getStateForBlockProduction(...)` directly.
> 
> Fork-choice/storage utilities are adjusted to support this: `ForkChoiceUtil` gains the committee-index hook, and `RecentChainData` inlines head `StateAndBlockSummary` retrieval instead of delegating to fork-specific logic.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1afcd13110be67993899d60b0c20d2e4b4eb823f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/AttestationUtil.java [hunk 1]
```diff
@@ -311,6 +313,14 @@ public AttestationData getGenericAttestationData(
   public abstract Attestation convertSingleAttestationToAggregated(
       BeaconState state, SingleAttestation singleAttestation);
 
+  public int computeCommitteeIndexForAttestation(
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/AttestationUtil.java]: It's not used anywhere and we have similar method in `ForkChoiceUtil`. What's the purpose of this one?

**Review by zilm13 — APPROVED** [review]
LGTM just one question

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/AttestationUtil.java (hunk 1 above)
Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/AttestationUtil.java]: Oh yeah forgot to remove that one, thought I can put it here first
